lib/containers/container_images.pm

```diff
@@ -118,7 +119,11 @@ sub test_opensuse_based_image {
 
     $version = 'Tumbleweed' if ($version =~ /^Staging:/);
 
-    if ($image_id =~ 'sles') {
+    if ($host_id =~ 'suse-microos') {
+        my ($sle_version, $sp_version) = get_sle_base($version);
+        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'grep PRETTY_NAME /etc/os-release' | cut -d= -f2",
```

Comment:
i would like to suggest the use of the existing solution for this. there is `check_os_release` and there is the `runtime` module. i found also [exec_on_container](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/container_images.pm#L179) (which might need some refactoring btw)

Reply:
For sure. The `get_sle_base` is definitely duplicate of `get_os_release` and there is also `check_os_release` as Yiannis said.